great_expectations/profiler/profiler_rule/profiler_rule.py

```diff
@@ -1,19 +1,27 @@
-from typing import Union
+from typing import Any, Dict, List, Optional, Union
 
-from ..core import ExpectationConfiguration
-from .configuration_builder.configuration_builder import ConfigurationBuilder
-from .domain_builder.domain_builder import DomainBuilder
-from .parameter_builder.parameter_builder import ParameterBuilder
-from .rule_state import RuleState
+from great_expectations.core import ExpectationConfiguration
+from great_expectations.execution_engine.execution_engine import MetricDomainTypes
+from great_expectations.profiler.configuration_builder.configuration_builder import (
+    ConfigurationBuilder,
+)
+from great_expectations.profiler.domain_builder.domain_builder import DomainBuilder
+from great_expectations.profiler.parameter_builder.parameter import Parameter
+from great_expectations.profiler.parameter_builder.parameter_builder import (
+    ParameterBuilder,
+)
+from great_expectations.profiler.profiler_rule.rule_state import RuleState
+from great_expectations.validator.validator import Validator
 
 
 class ProfilerRule:
     def __init__(
         self,
         name: str,
         domain_builder: DomainBuilder,
-        parameter_builders: [ParameterBuilder],
-        configuration_builders: [ConfigurationBuilder],
+        parameter_builders: List[ParameterBuilder],
+        configuration_builders: List[ConfigurationBuilder],
+        # TODO: <Alex>ALEX -- what should the type of "variables" be?</Alex>
```

Comment:
Based on your refactor of variables into a typed Parameter object, I would expect that variables would be a dictionary of Parameters.

Reply:
@talagluck I am seeing that `variables` is just a single `Parameter` object.  We should discuss.

Reply:
This is improved upon in PR #2718 /cc @talagluck @anthonyburdi